app/models/job.rb

```diff
@@ -55,11 +55,22 @@ def pct_format(number)
 
     def app_inspect(bin_count = 5, job_name = 'ondemand/sys/dashboard/sys/bc_osc_jupyter', cluster = '_all', property = 'CPUs')
       property_hash = Hash.new(0)
-      match_str = property == 'GPUs' ? /gpu:[^,]*(\d+)/ : /cpu=(\d*)/
-      Job.all.each do |job|
-        next unless job.name == job_name && job.tres && (cluster == '_all' || job.cluster == cluster)
+      if property == 'Hours'
+        Job.all.each do |job|
```

Comment:
Looks like an antipattern to initialize the hash outside the block.

You could do something like this instead:
```ruby
property_hash = Job.all.each_with_object(Hash.new(0)) do |job, property_hash|
                   if property == 'Hours'
                     # ...
                   else
                     # ...
                   end
                 end

```

Reply:
Ok, I didn't know you could do `each_with_object`. However, nesting the if inside of the loop seems like a losing tradeoff. I understand that this would make the code slightly simpler, but wouldn't it be better to have the if statements outside of the loop so that you would only have to check the property once instead of hundreds of times?

Reply:
> if statements outside of the loop so that you would only have to check the property once instead of hundreds of times?

Yea good catch, I hadn't through of the perfomance of the block.